app/services/sfn_pipeline_dispatch_service.rb

```diff
@@ -124,43 +109,41 @@ def convert_dag_json_to_wdl(dag_json)
     dag_tmp_file.unlink if dag_tmp_file
   end
 
-  def generate_wdl_input(stage_dags_json)
-    input_files_paths = @sample.input_files.map do |input_file|
-      File.join(@sample.sample_input_s3_path, input_file.name)
-    end
+  def generate_wdl_input
+    nucleotide_type = @sample.metadata.find_by(key: "nucleotide_type")
     sfn_pipeline_input_json = {
       Input: {
-        HostFilter: input_files_paths.each_with_index.map { |path, i| ["fastqs_#{i}", path] }.to_h,
+        fastqs_0: File.join(@sample.sample_input_s3_path, @sample.input_files[0].name),
+        fastqs_1: @sample.input_files[1] ? File.join(@sample.sample_input_s3_path, @sample.input_files[1].name) : nil,
+        star_genome: @sample.s3_star_index_path,
+        bowtie2_genome: @sample.s3_bowtie2_index_path,
+        max_fragments: @pipeline_run.max_input_fragments,
+        max_subsample_frag: @pipeline_run.subsample,
+        nucleotide_type: nucleotide_type ? nucleotide_type.string_validated_value : '',
+        human_star_genome: HostGenome.find_by(name: "Human").s3_star_index_path,
```

Comment:
Not sure if this is memoized but might make sense to load the host before since you use it twice.

Reply:
I think it is memoized in the Rails session.